pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/UnAckedMessageTracker.java

```diff
@@ -88,7 +90,7 @@ public void start(PulsarClientImpl client, ConsumerBase consumerBase, long ackTi
             public void run(Timeout t) throws Exception {
                 if (isAckTimeout()) {
                     log.warn("[{}] {} messages have timed-out", consumerBase, oldOpenSet.size());
-                    List<MessageIdImpl> messageIds = new ArrayList<>();
+                    Set<MessageIdImpl> messageIds = new HashSet<>();
```

Comment:
Messages already come from a HashSet, UnAckedTracker keeps sets of messages.
Converting them to Lists seems sort of a waste.